re_data_ui/src/pages/Dashboard.tsx

```diff
@@ -70,10 +80,17 @@ const formatOverviewData = (
       details.tableSchema.push(schema);
     } else if (item.type === 'test') {
       const schema = JSON.parse(item.data) as ITestSchema;
+      const run_at = dayjs(schema.run_at).format('YYYY-MM-DD HH:mm:ss') as string;
+
       schema.column_name = columnName;
       schema.model = model;
-      schema.run_at = dayjs(schema.run_at).format('YYYY-MM-DD HH:mm:ss');
-      details.testSchema.push(schema);
+      schema.run_at = run_at;
+
+      details.tests.push(schema);
+      runAtObject[run_at] = [...(runAtObject[run_at] || []), schema];
+      if (schema.status?.toLowerCase() === 'fail') {
+        failedTestsObject[model] = [...(failedTestsObject[model] || []), schema];
```

Comment:
why not pushing into the array using something like this below:
```js
if (failedTestsObject.hasOwnProperty(model)) {
    failedTestsObject[model].push(schema)
} else {
    failedTestsObject[model] = [schema]
}
```

the spread operator makes a copy each time leading to an O(N) operation while the `.push()` is amortized O(1).

Reply:
@dejii updated...